scripts/linters/pre_commit_linter.py

```diff
@@ -196,77 +197,71 @@ def _get_data(self, filepath, mode):
     """Run all lint checks.
 
     Args:
-        js_filepaths: list(str). The list of js filepaths to be linted.
-        ts_filepaths: list(str). The list of ts filepaths to be linted.
-        py_filepaths: list(str). The list of python filepaths to be linted.
-        html_filepaths: list(str). The list of HTML filepaths to be linted.
-        css_filepaths: list(str). The list of CSS filepaths to be linted.
+        file_extension_to_lint: str. The file extension to be linted.
         verbose_mode_enabled: bool. True if verbose mode is enabled.
-        file_extensions_to_lint: list(str). The list of file extensions to be
-            linted.
 
     Returns:
         custom_linter: list. Custom lint checks.
         third_party_linter: list. Third party lint checks.
     """
     parent_dir = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
-    config_path_for_css_in_html = os.path.join(
-        parent_dir, 'oppia', '.stylelintrc')
-    config_path_for_oppia_css = os.path.join(
-        parent_dir, 'oppia', 'core', 'templates', 'css', '.stylelintrc')
     custom_linters = []
     third_party_linters = []
 
-    js_ts_file_extension_type = 'js' in file_extensions_to_lint or (
-        'ts' in file_extensions_to_lint)
-    py_file_extension_type = 'py' in file_extensions_to_lint
-    html_file_extension_type = 'html' in file_extensions_to_lint
-    css_file_extension_type = 'css' in file_extensions_to_lint
-    other_file_extension_type = 'other' in file_extensions_to_lint
+    js_ts_file_extension_type = '.js' == file_extension_to_lint or (
+        '.ts' == file_extension_to_lint)
+    py_file_extension_type = '.py' == file_extension_to_lint
+    html_file_extension_type = '.html' == file_extension_to_lint
+    css_file_extension_type = '.css' == file_extension_to_lint
+    other_file_extension_type = 'other' == file_extension_to_lint
 
-    if not other_file_extension_type:
-        general_files_to_lint = '.' + file_extensions_to_lint
+    if js_ts_file_extension_type:
+        general_files_to_lint = _FILES['.js'] + _FILES['.ts']
+    elif other_file_extension_type:
+        general_files_to_lint = _FILES['other']
     else:
-        general_files_to_lint = file_extensions_to_lint
+        general_files_to_lint = _FILES['.%s' % file_extension_to_lint]
 
     custom_linter, third_party_linter = general_purpose_linter.get_linters(
-        _FILES[general_files_to_lint],
+        general_files_to_lint,
         verbose_mode_enabled=verbose_mode_enabled)
     custom_linters.append(custom_linter)
 
     if js_ts_file_extension_type:
         custom_linter, third_party_linter = js_ts_linter.get_linters(
-            js_filepaths, ts_filepaths,
+            _FILES['.js'], _FILES['.ts'],
             verbose_mode_enabled=verbose_mode_enabled)
         custom_linters.append(custom_linter)
         third_party_linters.append(third_party_linter)
 
     if html_file_extension_type:
```

Comment:
we are expecting to have only one file extension at a time right? In that case, use `elif`*

Reply:
Done.